### PR TITLE
Reg 128 add fees to price sheet product line

### PR DESCRIPTION
**Pull request by OksanaYeroshenko**

### CODE REVIEW

File: addons/regency_estimate/models/fee_value.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
+from odoo import fields, models, api, _
+
+
+class FeeValue(models.Model):
+    _name = 'fee.value'
+
+    name = fields.Char()
+    fee_type_id = fields.Many2one('fee.type')
```
Comment (edison12347) [inline addons/regency_estimate/models/fee_value.py:8]: can it be without type? if no, make it required

File: addons/regency_estimate/models/fee_value.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
+from odoo import fields, models, api, _
+
+
+class FeeValue(models.Model):
```
Comment (edison12347) [inline addons/regency_estimate/models/fee_value.py:4]: add _description

File: addons/regency_estimate/models/fee_type.py [hunk 3]
```diff
@@ -0,0 +1,8 @@
+from odoo import fields, models
+
+
+class FeeType(models.Model):
+    _name = 'fee.type'
```
Comment (edison12347) [inline addons/regency_estimate/models/fee_type.py:5]: add _description

File: addons/regency_estimate/models/fee_type.py [hunk 4]
```diff
@@ -0,0 +1,8 @@
...
+
+    name = fields.Char()
+    product_id = fields.Many2one('product.product', domain="[('detailed_type', '=', 'service')]")
```
Comment (edison12347) [inline addons/regency_estimate/models/fee_type.py:8]: add one2many field for fee value

Comment (edison12347) [inline addons/regency_estimate/models/fee_type.py:8]: is it required?

File: addons/regency_estimate/models/fee_value.py (hunk 1 above)
Comment (edison12347) [reply addons/regency_estimate/models/fee_value.py:8]: Think if any other field would be required here

File: addons/regency_estimate/models/product_price_sheet.py [hunk 5]
```diff
@@ -378,6 +408,11 @@ def onchange_price(self):
         for rec in self:
             rec.total = rec.price * rec.min_quantity
 
+    @api.depends('fee_value_ids', 'fee_value_ids.value', 'fee')
```
Comment (edison12347) [inline addons/regency_estimate/models/product_price_sheet.py:411]: missing price and min_qty

File: addons/regency_estimate/views/fee_value.xml [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+<odoo>
+
+    <record id="fee_value_list_view" model="ir.ui.view">
+        <field name="name">fee.value.tree</field>
+        <field name="model">fee.value</field>
+        <field name="arch" type="xml">
+            <tree string="Fee Types" editable="bottom">
+                <field name="name" string="Description"/>
+                <field name="fee_type_id"/>
+                <field name="value"/>
```
Comment (edison12347) [inline addons/regency_estimate/views/fee_value.xml:11]: It seems that either value or percent is required

**Review by edison12347 — CHANGES_REQUESTED** [review]

File: addons/regency_estimate/models/product_price_sheet.py [hunk 7]
```diff
@@ -378,6 +408,11 @@ def onchange_price(self):
         for rec in self:
             rec.total = rec.price * rec.min_quantity
 
+    @api.depends('fee_value_ids', 'fee_value_ids.value', 'fee', 'price', 'min_quantity')
```
Comment (artemShelest) [inline addons/regency_estimate/models/product_price_sheet.py:411]: 'fee_value_ids', 'fee_value_ids.value' dependencies are not needed here.

**Review by chupaSV — CHANGES_REQUESTED** [review]

File: addons/regency_estimate/models/fee_type.py [hunk 8]
```diff
@@ -0,0 +1,10 @@
+from odoo import fields, models
+
+
+class FeeType(models.Model):
+    _name = 'fee.type'
+    _description = 'Set additional fee types for setting additional fee values for price sheet lines'
```
Comment (edison12347) [inline addons/regency_estimate/models/fee_type.py:6]: _description - should be short - **Fee type** - this is how model will be named
adjust other _description as well

**Review by edison12347 — CHANGES_REQUESTED** [review]

**Review by chupaSV — APPROVED** [review]

**Review by edison12347 — APPROVED** [review]

**Pull request merged**